File: psana_examples/tags/V00-05-01/src/DumpL3T.cpp
//--------------------------------------------------------------------------
// File and Version Information:
// 	$Id$
//
// Description:
//	Class DumpL3T...
//
// Author List:
//      Andy Salnikov
//
//------------------------------------------------------------------------

//-----------------------
// This Class's Header --
//-----------------------
#include "psana_examples/DumpL3T.h"

//-----------------
// C/C++ Headers --
//-----------------

//-------------------------------
// Collaborating Class Headers --
//-------------------------------
#include "MsgLogger/MsgLogger.h"
#include "psddl_psana/l3t.ddl.h"

//-----------------------------------------------------------------------
// Local Macros, Typedefs, Structures, Unions and Forward Declarations --
//-----------------------------------------------------------------------

// This declares this class as psana module
using namespace psana_examples;
PSANA_MODULE_FACTORY(DumpL3T)

//		----------------------------------------
// 		-- Public Function Member Definitions --
//		----------------------------------------

namespace psana_examples {

//----------------
// Constructors --
//----------------
DumpL3T::DumpL3T (const std::string& name)
  : Module(name)
  , m_src()
{
  m_src = configSrc("source", "ProcInfo()");
}

//--------------
// Destructor --
//--------------
DumpL3T::~DumpL3T ()
{
}

/// Method which is called at the beginning of the run
void 
DumpL3T::beginRun(Event& evt, Env& env)
{
  shared_ptr<Psana::L3T::ConfigV1> config1 = env.configStore().get(m_src);
  if (config1) {
    WithMsgLog(name(), info, str) {
      str << "L3T::ConfigV1:";
      str << "\n  module_id = \"" << config1->module_id() << "\"";
      str << "\n  description = \"" << config1->desc() << "\"";
    }
  }
}

/// Method which is called with event data, this is the only required 
/// method, all other methods are optional
void 
DumpL3T::event(Event& evt, Env& env)
{
  // example of getting detector data from event
  shared_ptr<Psana::L3T::DataV1> data1 = evt.get(m_src);
  if (data1) {
    MsgLog(name(), info, "L3T::DataV1: accept = " << int(data1->accept()));
  }
}
  
} // namespace psana_examples

File: psana_examples/tags/V00-05-01/src/dump_l3t.py
#--------------------------------------------------------------------------
# File and Version Information:
#  $Id$
#
# Description:
#  Psana user analysis module dump_l3t...
#
#------------------------------------------------------------------------

"""Psana module to dump L3T information.

This software was developed for the LCLS project.  If you use all or 
part of it, please give an appropriate acknowledgment.

@version $Id$

@author Andy Salnikov
"""

#------------------------------
#  Module's version from SVN --
#------------------------------
__version__ = "$Revision$"
# $Source$

#--------------------------------
#  Imports of standard modules --
#--------------------------------
import sys
import logging

#-----------------------------
# Imports for other modules --
#-----------------------------
from psana import *

#----------------------------------
# Local non-exported definitions --
#----------------------------------

#---------------------
#  Class definition --
#---------------------
class dump_l3t (object) :
    '''Class whose instance will be used as a user analysis module.'''

    #----------------
    #  Constructor --
    #----------------
    def __init__(self):
        '''Class constructor. Does not need any parameters'''

        self.m_src = self.configSrc('source', 'ProcInfo()')

    #-------------------
    #  Public methods --
    #-------------------

    def beginRun( self, evt, env ) :
        """This optional method is called if present at the beginning 
        of the new run.

        @param evt    event data object
        @param env    environment object
        """

        config = env.configStore().get(L3T.Config, self.m_src)
        if config:
        
            print "%s: %s" % (config.__class__.__name__, self.m_src)
            print "  module_id =\"{0}\"".format(config.module_id())
            print "  description =\"{0}\"".format(config.desc())

    def event( self, evt, env ) :
        """This method is called for every L1Accept transition.

        @param evt    event data object
        @param env    environment object
        """

        data = evt.get(L3T.Data, self.m_src)
        if data:
            print "{0}: accept={1}".format(data.__class__.__name__, data.accept())

